### PR TITLE
Lexer tests

### LINKED ISSUE #10: Lexer: Tests for unexpected behaviour

### LINKED ISSUE #12: Lexer: Clean up of existing code

### LINKED ISSUE #35: Lexer fails to match a negative float, or numbers with invalid syntax.
related #10

### PR DESCRIPTION
**Pull request by Dreamonic**
Add more tests for the lexer

Closes #10 
Closes #12 
Closes #35

### CODE REVIEW

File: test/unit/LexerSpec.hs [hunk 1]
```diff
@@ -2,33 +2,88 @@ module LexerSpec (spec) where
...
+
+    --- tokenize ---
+
+tsTokenizeAllModule = it "Lexing a module" $
+    tokenize "(module (memory 1) (func))" `shouldBe`
+    [LP, Keyword "module", LP, Keyword "memory", UnsignedN 1, RP, LP, Keyword "func", RP, RP]
+
+tsTokenizeAllFunction = it "Lexing a function" $
+    tokenize "(func (param i32) (param f32) (local f64)\nget_local 0\nget_local 1\nget_local 2)" `shouldBe`
+    [LP, Keyword "func", LP, Keyword "param", Keyword "i32", RP, LP, Keyword "param", Keyword "f32",RP,LP,Keyword "local",Keyword "f64",RP,Keyword "get_local",UnsignedN 0, Keyword "get_local",UnsignedN 1, Keyword "get_local",UnsignedN 2,RP]
```
Comment (SiertSalabim) [inline test/unit/LexerSpec.hs]: Line breaks would be nice maybe.

File: src/LexerRegex.hs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+tokenizeS token 
+    | token == "(" = LP
+    | token == ")" = RP
+    | token =~ ("^\\$(" ++ idcharRegex ++ ")+$") = ID $ tail token
+    | token =~ "^\\\"(.|\n)*?\\\"$" = Str $ tail $ init token
+    | token =~ ("^[a-z](" ++ idcharRegex ++ ")*$") = Keyword token
+    | token =~ "^[0-9]+$" = UnsignedN (read token :: Integer)
+    | token =~ "^-[0-9]+$" = SignedN (read token :: Integer)
+    | token =~ "^-?[0-9.]+$" = FloatN (read token :: Double)
+    | otherwise = Reserved token
```
Comment (SiertSalabim) [inline src/LexerRegex.hs]: There is some documentation missing here.

Comment (SiertSalabim) [reply src/LexerRegex.hs]: Otherwise, looks very good.

File: test/unit/LexerSpec.hs (hunk 1 above)
Comment (Dreamonic) [reply test/unit/LexerSpec.hs]: Done

File: src/LexerRegex.hs (hunk 2 above)
Comment (Dreamonic) [reply src/LexerRegex.hs]: Added a bit more detailed docs as well :P

**Comment by nickyu42** [comment]
All of the changes look good, can be merged

**Pull request merged**